src/mono/sample/wasm/browser-bench/AppStart.cs

```diff
@@ -0,0 +1,88 @@
+        static Type jsRuntimeType = System.Type.GetType("System.Runtime.InteropServices.JavaScript.Runtime, System.Private.Runtime.InteropServices.JavaScript", true);
+        static Type jsFunctionType = System.Type.GetType("System.Runtime.InteropServices.JavaScript.Function, System.Private.Runtime.InteropServices.JavaScript", true);
+        [DynamicDependency("InvokeJS(System.String)", "System.Runtime.InteropServices.JavaScript.Runtime", "System.Private.Runtime.InteropServices.JavaScript")]
+        static MethodInfo invokeJSMethod = jsRuntimeType.GetMethod("InvokeJS", new Type[] { typeof(string) });
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicConstructors, "System.Runtime.InteropServices.JavaScript.Function", "System.Private.Runtime.InteropServices.JavaScript")]
+        static ConstructorInfo functionConstructor = jsRuntimeType.GetConstructor(new Type[] { typeof(object[]) });
+        [DynamicDependency("Call()", "System.Runtime.InteropServices.JavaScript.Function", "System.Private.Runtime.InteropServices.JavaScript")]
+        static MethodInfo functionCall = jsFunctionType.GetMethod("Call", BindingFlags.Instance | BindingFlags.Public, new Type[] { });
+
+        public AppStartTask()
+        {
+            measurements = new Measurement[] {
+                new PageShow(),
+                new ReachManaged(),
+            };
+        }
+
+        Measurement[] measurements;
+        public override Measurement[] Measurements => measurements;
+
+        static string InvokeJS(string js)
+        {
+            return (string)invokeJSMethod.Invoke(null, new object[] { js });
+        }
+
+        class PageShow : BenchTask.Measurement
+        {
+            public override string Name => "Page show";
+
+            public override int InitialSamples => 3;
+
+            async Task RunAsyncStep()
+            {
+                var function = Activator.CreateInstance(jsFunctionType, new object[] { new object[] { @"return App.StartAppUI();" } });
+                var task = (Task<object>)functionCall.Invoke(function, new object[] { });
+
+                await task;
+            }
+
+            public override bool HasRunStepAsync => true;
+
+            public override async Task RunStepAsync()
+            {
+                var function = Activator.CreateInstance(jsFunctionType, new object[] { new object[] { @"return App.PageShow();" } });
+                await (Task<object>)functionCall.Invoke(function, null);
+            }
+        }
+
+        class ReachManaged : BenchTask.Measurement
+        {
+            public override string Name => "Reach managed";
+            public override int InitialSamples => 3;
+            public override bool HasRunStepAsync => true;
+
+            static object jsUIReachedManagedFunction = Activator.CreateInstance(jsFunctionType, new object[] { new object[] { @"return App.ReachedManaged();" } });
```

Comment:
This could have variants with different number of parameters, types of parameters marshalled and return type.

Reply:
Here it is simple parameter less function, so hopefully quite fast. The idea about marshalling variance is cool, maybe we can add new task to measure various marshaling scenarios and features.